sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -2649,21 +2649,36 @@ def _AddPartitionKey(self, collection_link, document, options):
 
     # Extracts the partition key from the document using the partitionKey definition
     def _ExtractPartitionKey(self, partitionKeyDefinition, document):
+        if partitionKeyDefinition["kind"] == "MultiHash":
+            ret = []
+            for partition_key_level in partitionKeyDefinition.get("paths"):
+                # Parses the paths into a list of token each representing a property
+                partition_key_parts = base.ParsePaths([partition_key_level])
+                # Check if the partitionKey is system generated or not
+                is_system_key = partitionKeyDefinition["systemKey"] if "systemKey" in partitionKeyDefinition else False
+
+                # Navigates the document to retrieve the partitionKey specified in the paths
+                val = self._retrieve_partition_key(partition_key_parts, document, is_system_key)
+                if val is _Undefined:
+                    raise ValueError("Undefined Value in MultiHash PartitionKey.")
```

Comment:
same here, let's let the service take care of this when we send an incomplete definition as opposed to assigning undefined values to check within the client - same for the async version

Reply:
This is something we discussed. If an undefined value is passed as a partition key, it doesn't get sent to the backend because it causes a type error because the undefined value is not json serializable. This is a specific edge case. We also don't want to pass undefined values in a multi hash partition key compared to single hash, so then we either leave the type error or we create a value error with a message that informs the user what the error is.